lib/aws/google/cached_credentials.rb

```diff
@@ -17,28 +17,45 @@ def initialize(options = {})
 
         @profile = options[:profile] || ENV['AWS_PROFILE'] || ENV['AWS_DEFAULT_PROFILE'] || 'default'
         @session_profile = @profile + '_session'
-        @expiration = Aws.shared_config.expiration(profile: @session_profile) rescue nil
-        @credentials = Aws.shared_config.credentials(profile: @session_profile) rescue nil
+        @expiration = begin
+          Aws.shared_config.expiration(profile: @session_profile)
+        rescue StandardError
+          nil
+        end
+        @credentials = begin
+          Aws.shared_config.credentials(profile: @session_profile)
+        rescue StandardError
+          nil
+        end
         refresh_if_near_expiration
       end
 
       def refresh_if_near_expiration
-        if near_expiration?(SYNC_EXPIRATION_LENGTH)
-          @mutex.synchronize do
-            if near_expiration?(SYNC_EXPIRATION_LENGTH)
-              refresh
-              write_credentials
-            end
+        return unless near_expiration?(SYNC_EXPIRATION_LENGTH)
+
+        @mutex.synchronize do
+          if near_expiration?(SYNC_EXPIRATION_LENGTH)
+            refresh
+            write_credentials
           end
         end
       end
 
       # Write credentials and expiration to AWS credentials file.
       def write_credentials
-        # AWS CLI is needed because writing AWS credentials is not supported by the AWS Ruby SDK.
+        # Ensure the AWS CLI is available before attempting to write credentials.
         return unless system('which aws >/dev/null 2>&1')
-        Aws::SharedCredentials::KEY_MAP.transform_values(&@credentials.method(:send)).
-          merge(expiration: @expiration).each do |key, value|
+
+        # Manually map the credentials to the keys used by AWS CLI
+        credentials_map = {
+          'aws_access_key_id' => @credentials.access_key_id,
+          'aws_secret_access_key' => @credentials.secret_access_key,
+          'aws_session_token' => @credentials.session_token,
+          'expiration' => @expiration
+        }
+
+        # Use the AWS CLI to set the credentials in the session profile
+        credentials_map.each do |key, value|
           system("aws configure set #{key} #{value} --profile #{@session_profile}")
         end
```

Comment:
This code had to be modified to work with the new version of `aws-sdk-core`. The `KEY_MAP` constant does not exist in the latest version.

Reply:
I kind of like explicitly listing them like this anyway anyway, this replaces fairly complex VERYSMARTCODE with a few lines of very clear code 💗